File: main/java/nullbots/nullmage/IsCurseSelected.java
package nullbots.nullmage;

import com.runemate.game.api.osrs.local.hud.interfaces.Magic;
import com.runemate.game.api.script.framework.tree.BranchTask;
import com.runemate.game.api.script.framework.tree.TreeTask;

public class IsCurseSelected extends BranchTask {

    private ChooseSpell chooseSpell = new ChooseSpell();
    private CurseBear curseBear = new CurseBear();

    @Override
    public TreeTask successTask() {
        return curseBear;
    }

    @Override
    public TreeTask failureTask() {
        return chooseSpell;
    }

    @Override
    public boolean validate() {
        return Magic.STUN.isSelected();
    }
}

File: kotlin/archbots/archmage/treebot/branches/IsCurseSelected.kt
package archbots.archmage.treebot.branches

import archbots.archmage.treebot.leaves.CurseBear
import com.runemate.game.api.osrs.local.hud.interfaces.Magic
import archapi.treebot.framework.BranchTask
import archapi.treebot.framework.TreeTask

class IsCurseSelected : BranchTask() {

    private val chooseSpell = ChooseSpell()
    private val curseBear = CurseBear()

    override fun successTask(): TreeTask {
        return curseBear
    }

    override fun failureTask(): TreeTask {
        return chooseSpell
    }

    override fun validate(): Boolean {
        return Magic.STUN.isSelected
    }
}
